### PR TITLE
core(runner): split lhr, artifacts return, respect output type

### PR DESCRIPTION
**Pull request by patrickhulce**
jumped on @brendankenny's suggestion to make `{lhr, artifacts}` and a breaking feature in the process (#4333), having node module respect the `output=html` setting

also I had a bug in config with default settings when undefined (mostly happens in tests, but could be user explicitly passing undefined) so fixing that was good 👍

### CODE REVIEW

**Comment by brendankenny** [comment]
also, FWIW, "core(runner): respect output setting" is definitely not the most important change happening here :P

**Comment by patrickhulce** [comment]
> also, FWIW, "core(runner): respect output setting" is definitely not the most important change happening here :P

what's the most important change?

> probably easier to still have the report generated outside of runner so that the devtools bundle doesn't need it 

I see, so against the feature existing or just proposing index.js?

**Comment by patrickhulce** [comment]
> also the lighthouse-background side will currently be broken

yes this was meant to be proposal, added [WIP]

**Comment by paulirish** [comment]
One option: add browserify ignore for lib/format-output.. and always check if it's truth before using it. But I should look closer before jumping to a solution

**Comment by brendankenny** [comment]
> what's the most important change?

splitting up the return value from the module :)

> I see, so against the feature existing or just proposing index.js?

oh, I think 👍 👍, just `index.js` or have to do something about the bundling

**Comment by patrickhulce** [comment]
aight folks ready for re-review!

File: lighthouse-core/report/v2/report-formatter.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+'use strict';
+
+const ReportGenerator = require('./report-generator');
+
+module.exports = {
+  /**
+   * Creates the results output in a format based on the `mode`.
+   * @param {LH.Result} lhr
+   * @param {string} outputMode
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-core/report/v2/report-formatter.js]: done

**Comment by patrickhulce** [comment]
> do we want to refer to the final report string as the formattedReport instead of the generatedReport? That makes a certain amount of sense, but I'd like to get rid of the one we don't use because they sound very similar (e.g. if we use formattedReport, there shouldn't be a report-generator.js

resolved on `.report` 🎉 

> one way of doing this would be to merge report-formatter.js and report-generator.js in this PR, a single file that makes reports, but kick out the fs.readFileSync() calls and reportJs/reportCss/reportTemplates getters to a different file that's just like html-report-assets.js or whatever and we can browserify.ignore() that file

great idea, done! 👍

**Comment by ebidel** [comment]
It's a party.

**Pull request merged**